aws-sdk-resources/lib/aws-sdk-resources/services/s3/file_downloader.rb

```diff
@@ -135,31 +133,43 @@ def map_files(slice, dir, mode)
         end
       end
 
-      def get_range(chunks)
+      def clean_up_parts(parts)
+        parts.each do |filename|
+          File.unlink(filename) if File.exists?(filename)
+        end
+      end
+
+      def multithreaded_get_by_ranges(chunks)
         thread_batches(chunks, 'range')
       end
 
-      def get_part(parts)
+      def multithreaded_get_by_parts(parts)
         thread_batches(parts, 'part_number')
       end
 
       def thread_batches(chunks, param)
         batches = file_batches(chunks, param)
+        parts = batches.flat_map(&:values)
         batches.each do |batch|
           threads = []
-          batch.each do |chunk, file|
-            threads << Thread.new do
-              resp = @client.get_object(
-                :bucket => @bucket,
-                :key => @key,
-                param.to_sym => chunk,
-                :response_target => file
-              )
+          begin
+            batch.each do |chunk, file|
+              threads << Thread.new do
+                resp = @client.get_object(
+                  :bucket => @bucket,
+                  :key => @key,
+                  param.to_sym => chunk,
+                  :response_target => file
+                )
+              end
             end
+            threads.each(&:join)
+          rescue => error
```

Comment:
Given that we're rescuing all of StandardError, perhaps we could simplify this to be an `ensure` block, which means we only ever need to call `#clean_up_parts` here whether the method succeeds or fails. It also eliminates the need to catch only to re-raise the same exception.